app/core/RPCMethods/RPCMethodMiddleware.ts

```diff
@@ -0,0 +1,480 @@
+					params.from = firstParam;
+				}
+
+				const pageMeta = {
+					meta: {
+						url: url.current,
+						title: title.current,
+						icon: icon.current,
+					},
+				};
+				const rawSig = await PersonalMessageManager.addUnapprovedMessageAsync({
+					...params,
+					...pageMeta,
+				});
+
+				res.result = rawSig;
+			},
+
+			eth_signTypedData: async () => {
+				const { TypedMessageManager } = Engine.context;
+				const pageMeta = {
+					meta: {
+						url: url.current,
+						title: title.current,
+						icon: icon.current,
+					},
+				};
+				const rawSig = await TypedMessageManager.addUnapprovedMessageAsync(
+					{
+						data: req.params[0],
+						from: req.params[1],
+						...pageMeta,
+					},
+					'V1'
+				);
+
+				res.result = rawSig;
+			},
+
+			eth_signTypedData_v3: async () => {
+				const { TypedMessageManager } = Engine.context;
+				const data = JSON.parse(req.params[1]);
+				const chainId = data.domain.chainId;
+
+				const {
+					provider: { type: networkType },
+					network,
+				} = Engine.context.NetworkController.state;
+
+				const activeChainId = networkType === RPC ? network : Networks[networkType].networkId;
+				// eslint-disable-next-line
+				if (chainId && chainId != activeChainId) {
+					throw ethErrors.rpc.invalidRequest(
+						`Provided chainId (${chainId}) must match the active chainId (${activeChainId})`
+					);
+				}
+
+				const pageMeta = {
+					meta: {
+						url: url.current,
+						title: title.current,
+						icon: icon.current,
+					},
+				};
+
+				const rawSig = await TypedMessageManager.addUnapprovedMessageAsync(
+					{
+						data: req.params[1],
+						from: req.params[0],
+						...pageMeta,
+					},
+					'V3'
+				);
+
+				res.result = rawSig;
+			},
+
+			eth_signTypedData_v4: async () => {
+				const { TypedMessageManager } = Engine.context;
+				const data = JSON.parse(req.params[1]);
+				const chainId = data.domain.chainId;
+
+				const {
+					provider: { type: networkType },
+					network,
+				} = Engine.context.NetworkController.state;
+
+				const activeChainId = networkType === RPC ? network : Networks[networkType].networkId;
+
+				// eslint-disable-next-line eqeqeq
+				if (chainId && chainId != activeChainId) {
+					throw ethErrors.rpc.invalidRequest(
+						`Provided chainId (${chainId}) must match the active chainId (${activeChainId})`
+					);
+				}
+
+				const pageMeta = {
+					meta: {
+						url: url.current,
+						title: title.current,
+						icon: icon.current,
+					},
+				};
+				const rawSig = await TypedMessageManager.addUnapprovedMessageAsync(
+					{
+						data: req.params[1],
+						from: req.params[0],
+						...pageMeta,
+					},
+					'V4'
+				);
+
+				res.result = rawSig;
+			},
+
+			web3_clientVersion: async () => {
+				let version = appVersion;
+				if (!version) {
+					appVersion = await getVersion();
+					version = appVersion;
+				}
+				res.result = `MetaMask/${version}/Beta/Mobile`;
```

Comment:
could this instead be?
I don't think you need a new variable?
```suggestion
				if (!appVersion) {
					appVersion = await getVersion();
				}
				res.result = `MetaMask/${version}/Beta/Mobile`;
```

Reply:
Should this `MetaMask/${version}/Beta/Mobile;` also be changed?

Reply:
I have updated to use Ricky code. Regarding Beta/Mobile I think we could change it but no right now because I am not sure if devs are using this specific string to check if it's the mobile app using their dapp.

Reply:
I updated Beta/Mobile to just Mobile. I just saw that the text changed with {version} so devs should be expecting it to change always.